packages/devtools_app/lib/src/shared/routing.dart

```diff
@@ -205,8 +236,13 @@ class DevToolsRouterDelegate extends RouterDelegate<DevToolsRouteConfiguration>
 
     final currentPage = currentConfiguration!.page;
     final newArgs = {...currentConfiguration!.args, ...argUpdates};
-    _replaceStack(DevToolsRouteConfiguration(currentPage, newArgs));
-    notifyListeners();
+    _replaceStack(
+      DevToolsRouteConfiguration(
+        currentPage,
+        newArgs,
+        currentConfiguration!.state,
```

Comment:
pull currentConfifuration out into a var before currentPage so it only has to be null asserted once